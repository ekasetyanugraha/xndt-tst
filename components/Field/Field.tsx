import React from 'react';

interface Props {
  label?: string;
  help?: string;
  children?: JSX.Element | JSX.Element[];
}

export default function Field({ label, help, children }: Props): JSX.Element {
  return (
    <div className="field" data-testid="field">
      {label && <label className="label">{label}</label>}
      {children && <div className="control">{children}</div>}
      {help && <p className="help">{help}</p>}
    </div>
  );
}
