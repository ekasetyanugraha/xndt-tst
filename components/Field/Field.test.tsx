import React from 'react';
import { render, screen } from '@testing-library/react';
import Field from '.';

describe('Component: Field', () => {
  it('should render properly without any props', () => {
    render(<Field />);

    expect(screen.getByTestId('field')).toMatchSnapshot();
  });

  it('should render properly', () => {
    render(
      <Field label="Test Label" help="Test Help">
        <p>Test</p>
      </Field>,
    );

    expect(screen.getByTestId('field')).toMatchSnapshot();
  });
});
