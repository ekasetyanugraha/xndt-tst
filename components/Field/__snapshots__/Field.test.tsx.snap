// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Field should render properly 1`] = `
<div
  class="field"
  data-testid="field"
>
  <label
    class="label"
  >
    Test Label
  </label>
  <div
    class="control"
  >
    <p>
      Test
    </p>
  </div>
  <p
    class="help"
  >
    Test Help
  </p>
</div>
`;

exports[`Component: Field should render properly without any props 1`] = `
<div
  class="field"
  data-testid="field"
/>
`;
